src/inmanta/module.py

```diff
@@ -268,7 +271,79 @@ def merge_specs(mainspec: "Dict[str, List[Requirement]]", new: "List[Requirement
             mainspec[key] = mainspec[key] + [req]
 
 
-class ModuleLike(object):
+class InstallMode(str, enum.Enum):
+    release = "release"
+    prerelease = "prerelease"
+    master = "master"
+
+
+class FreezeOperator(str, enum.Enum):
+    eq = "=="
+    compatible = "~="
+    ge = ">="
+
+    @classmethod
+    def get_regex_for_validation(cls) -> str:
+        all_values = [o.value for o in FreezeOperator]
+        return f"^({'|'.join(all_values)})$"
+
+
+class Metadata(BaseModel):
+    name: str
+    description: Optional[str] = None
+    requires: List[str] = []
+    freeze_recursive: bool = False
+    freeze_operator: str = Field(default="~=", regex=FreezeOperator.get_regex_for_validation())
```

Comment:
Not sure how feasible but it would be nice to have the docstring with parameters here and just load that from sphinx with autodoc. Feel free to ignore this comment if you think it wouldn't be worth the effort.